README.md

```diff
@@ -349,8 +349,6 @@ The runtime expression makes use of [string-json-template-language](https://gith
 
 The table below provides examples of runtime expressions and examples of their use in a value:
 
-Runtime expressions preserve the type of the referenced value.
```

Comment:
I don't think we should preserve type. I think it should always be strings. I would argue that this would be the syntax if you need to have an object or array in a link:

given:
```
{
  "result": {
    "pet": {
      "id": 1,
      "name": "foofoo"
    }
  }
}
```


**this is what we should only allow**
```
{
  "links": [
    {
      "name": "myLink",
      "method": "other_method",
      "params": {
        "pet": {
          "id": "${result.id}",
          "name": "${result.name}"
        }
      }
    }
  ]
}
```


where as if you passed by referenced value, we are **allowing this**:


```
{
"links": [
  {
    "name": "myLink",
    "method": "other_method",
    "params": {
      "pet": "${result}"
    }
  }
]
}
```

Reply:
another way to solve this is to say that there must not be anything in the `head` or `tail` of the `template` to be able to pass by value.

ex:

**not allowed**
`"foo${result}"`

**allowed**
`"${result}"`

Reply:
just to note that you **cant** actually do this in openapi, theres an open bug ticket to pass array values

Reply:
```json
{
  "links": [
    {
      "name": "myLink",
      "method": "other_method",
      "params": {
        "pet": {
          name: "${result.name}",
          ...
        }
      }
    }
  ]
}
```

Reply:
what about arrays? where you cant just do 1 at a time, its dynamic:

```
{
  "links": [
    {
      "name": "myLink",
      "method": "other_method",
      "params": {
        "pet": {
          "id": "${result.id}",
          "names": [
            "${result.names[0]}",
            "${result.names[1]}",
            "${result.names[2]}"
          ]
        }
      }
    }
  ]
}```